src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/bolts/IslCacheBolt.java

```diff
@@ -120,15 +131,34 @@ protected void handleInput(Tuple input) throws PipelineException {
 
             emit(ACTION_STREAM_ID.name(), input, new Values(flowId, direction, latency, maxLatency, maxLatencyTier2,
                     getCommandContext()));
+            emitLatencyStats(input, flowId, direction, latency);
         } else {
             unhandledInput(input);
         }
     }
 
+    private void emitLatencyStats(Tuple input, String flowId, FlowDirection direction, long latency) {
+        Map<String, String> tags = ImmutableMap.of(
+                "flowid", flowId,
+                "direction", direction.name().toLowerCase(),
+                "calculated", "true"
+        );
+
+        Datapoint datapoint = new Datapoint(metricFormatter.format("flow.rtt"),
+                System.currentTimeMillis(), tags, latency);
```

Comment:
We should not tag metric point with current time (due to queue on input i.e. we can read data with some significant delay). We can read timestamp from input message or we can add timestamp field in payload.